### ISSUE TITLE
SSL "Fix"" Breaks MIM-H02

### ISSUE THREAD

**Issue by lawrencelnm** [atxbyea/samsungrac#52]
The latest release 4.2 / 4.1 which targets SSL fix broke the integration with MIM-H02 (Using 'samsung_2878.yaml')

As soon as I roll back to 4.0, the air con shows up again in Home Assistant.

**Comment by smartekIT** [comment]
I agree with you, but v4.1 is working. v4.2 breaks it again.

**Comment by smartekIT** [comment]
so for now whoever using MIM-H02 like me and you should stay on v4.1 until they fix v4.2 to be compatible with MIM-H02

**Comment by lawrencelnm** [comment]
Thanks for the clarification. I didn’t manage to go onto 4.1 as it moved me from 4.0 straight to 4.2.

**Comment by thomas70** [comment]
My stopped working after upgrade to 2022.11.5 i think. No versions is working

**Comment by catatonico2** [comment]
same here, no version works

**Comment by buzurk26** [comment]
Yeah mines stopped working now on any version, seems to have dropped off in recent updates of HA

**Comment by rgogada** [comment]
so what are you guys (with MM02 - samsung_2878.yaml) are using?

**Comment by lawrencelnm** [comment]
I stayed on 4.0 / 4.1. That the version I know works with MIM H02

**Comment by buzurk26** [comment]
> I stayed on 4.0 / 4.1. That the version I know works with MIM H02

Yep same here, seems to work on those versions.

**Comment by rgogada** [comment]
that is what I did too until yesterday. But the new 2023.6 beta broken it this morning. I can roll back to regular release for now but it may still broken when they release it next week.

**Comment by lawrencelnm** [comment]
> that is what I did too until yesterday. But the new 2023.6 beta broken it this morning. I can roll back to regular release for now but it may still broken when they release it next week.

Really? Damn. That’s no good !

**Comment by rgogada** [comment]
does anyone know if MIM-H03 or MIM-H04 work reliably? Happy to throw couple of hundreds at this problem to go away instead of mucking around it for each update. Thanks for your inputs.

**Comment by rgogada** [comment]
> I see the integration has a V5.0. Does that work with HA 2023.6 + MIM-H02 ?

tried that. no dice :(

**Comment by lawrencelnm** [comment]
> does anyone know if MIM-H03 or MIM-H04 work reliably? Happy to throw couple of hundreds at this problem to go away instead of mucking around it for each update. Thanks for your inputs.

Originally I got MIM-H02 to work. Then one of the samsungrac integration release broke MIM-H02. 

I threw in the towel, and bought a MIM-H03, which was still compatible with my AC. 
I had lots of issue with MIM-H03 as well. I can't remember the issues now as it's been a few years.

I reverted back to MIM-H02 after one of the integration samsungrac release got it working again.

**Comment by rgogada** [comment]
weird thing is I have reverted to HA 2023.5 and  climate_ip 4.1 still getting the same error. I have restored from last night's backup still same error.  just to make sure I restarted my whole aircon system (including the MIM-H02) and vola.. its working. atleast with 4.1 for now.

**Comment by lawrencelnm** [comment]
> weird thing is I have reverted to HA 2023.5 and climate_ip 4.1 still getting the same error. I have restored from last night's backup still same error. just to make sure, I restarted my whole aircon system (including the MIM-H02) and vola.. its working. atleast with 4.1 for now.

So are you back on 2023.5 now with 4.1 ?

**Comment by lawrencelnm** [comment]
Hi all. I saw Release 5.2 was out, but there wasn't much detailed in the release note other than another 'SSL' fix. I decided to bite the bullet and test this out.

I am happy to report that I have successfully upgraded from Release 4.1 to release 5.2 without breaking MIM-H02.
Other than that, I have also (finally) managed to move off Home Assistant 2023.5 and onto the latest 2023.9.2, running this new release 5.2. MIM-H02 is still operational.

For those who has been stuck (in the past release), we have now able to move forwards.
Thank you to whoever that made MIM-H02 compatible again with Home Assistant !
